### ISSUE TITLE
Module Reorganization

### ISSUE THREAD

**Issue by QuentinRoy** [leongersen/noUiSlider#794]
I just would like to let you know that as I newcomer trying to make my way in the library's codebase, I often found it easier and more efficient to look at the compiled file then try to find back what I am interested in the source code than working with the source code. I lost a significant amount of time trying to find where is what.
This seems to indicate that the division of the codebase fails in his job to make it more apprehensible (I humbly feel it may actually make it worse than no division at all). It seems I am not the only one who experienced it (#766). The current division might do a great job of making developers familiar with the codebase more productive though, this I cannot tell.

**Comment by leongersen** [comment]
Thanks for your comments on this, I agree with you here. The idea for the separation was that everything sharing a state is grouped in `scope_` files, and everything that's not privileged to this state is in a separately named file. The problem, I think, is that there's no clear idea about what code belongs in which `scope_` file. Merging those files would probably already help to get a better idea of the codebase.

**Comment by QuentinRoy** [comment]
I suppose you already considered using "real" modules (i.e. esm or commonjs) and relying on a bundler like rollup?

**Comment by leongersen** [comment]
I did briefly, but any modularization approach would run into the same
issues regarding separation of concepts. I don't see bringing in additional
tooling easily fixing this.

**Comment by QuentinRoy** [comment]
I do not pretend it will fix it, but I find it does help a bit in finding where variables come from (because they need to be imported to be used) and understand what depends on what. It also (generally) exposes a clear starting point (i.e. index.js) and usually takes care of the "stitching" without you needing to add it in the code base (e.g. the UMD wrapper or `scope_end.js`). And finally, people are used to it now... These may be little things but they're worse what they're worse..

**Comment by leongersen** [comment]
I've just pushed a refactor that should make it a little easier to work on the source as a beginner. I've moved most functions to the file where they are called, and split out functionality related to event binding and the external API. I considered modularizing more, but the current source isn't really designed for that, so I let it go.

Anyway, anyone looking to make changes now has two files to look at:
`scope_core`: I've cut this down to ~200 lines. This file contains the logic regarding positioning and movement of the slider handles;
`scope_events`: Implements handling of browser events and plumbs them to the relevant actions from the `core` file. 

Thanks for helping out, @QuentinRoy.
